services/auth/src/api.js

```diff
@@ -265,48 +261,40 @@ builder.declare({
   await req.authorize({clientId, scopes});
 
   let accessToken = slugid.v4() + slugid.v4();
-  let client = await this.Client.create({
-    clientId: clientId,
-    description: input.description,
-    accessToken: accessToken,
-    expires: new Date(input.expires),
-    scopes: scopes,
-    disabled: 0,
-    details: {
-      created: new Date().toJSON(),
-      lastModified: new Date().toJSON(),
-      lastDateUsed: new Date().toJSON(),
-      lastRotated: new Date().toJSON(),
-      deleteOnExpiration: !!input.deleteOnExpiration,
-    },
-  }).catch(async (err) => {
-    // Only handle
-    if (err.code !== 'EntityAlreadyExists') {
+
+  try {
+    await this.db.fns.create_client(
+      clientId,
+      input.description,
+      this.db.encrypt({value: Buffer.from(accessToken, 'utf8')}),
+      new Date(input.expires),
+      false,
+      JSON.stringify(scopes),
+      !!input.deleteOnExpiration,
+    );
+  } catch (err) {
+    if (err.code !== UNIQUE_VIOLATION) {
       throw err;
     }
 
-    // Load client
-    let client = await this.Client.load({clientId});
+    // Load existing client
+    let [client] = await this.db.fns.get_client(clientId);
+    if (!client) {
+      throw err; // exists, or doesn't exist??
+    }
 
     // If stored client different or older than 15 min we return 409
-    let created = new Date(client.details.created).getTime();
+    let created = new Date(client.created).getTime();
 
     if (client.description !== input.description ||
         client.expires.getTime() !== new Date(input.expires).getTime() ||
         !_.isEqual(client.scopes, scopes) ||
-        client.disabled !== 0 ||
+        client.disabled ||
         created < Date.now() - 15 * 60 * 1000) {
       return res.reportError('RequestConflict',
         'client with same clientId already exists, possibly an issue with retry logic or idempotency',
         {});
```

Comment:
this is fine but part of me feels like this could be an upsert with all of the logic here in the db instead. would that even be desirable?